src/state.c

```diff
@@ -403,6 +405,46 @@ handle_state_proxy_section(GNode* node, struct handler_data* data) {
 }
 
 /*!
+ * handler for pod section.
+ *
+ * \param node \c GNode.
+ * \param data \ref handler_data.
+ */
+static void
+handle_state_pod_section(GNode* node, struct handler_data* data) {
+	struct cc_pod *pod;
+
+	if (! (node && node->data)) {
+		return;
+	}
+	if (! (node->children && node->children->data)) {
+		g_critical("%s missing value", (char*)node->data);
+		return;
+	}
+
+	g_assert (data->state);
```

Comment:
No check on `data` before dereferencing. Also, an we convert the assert to a test and call to `g_critical()`? (We're trying to phase out asserts wherever possible).